apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/tasks/OperationTask.java

```diff
@@ -140,42 +152,43 @@ private Optional<JobState> submitJob(Attributes typeAttributes) {
 
   private Optional<JobState> pollJobStatus(Attributes typeAttributes) {
     long startTime = System.currentTimeMillis();
-    while (!jobFinished()) {
-      long elapsedTime = System.currentTimeMillis() - startTime;
-      if (elapsedTime > timeoutMs) {
+    try {
+      while (!jobFinished()) {
+        long elapsedTime = System.currentTimeMillis() - startTime;
         Optional<JobState> job = jobsClient.getState(jobId);
-        // Do not cancel job if it is still in running state after 3 hours
-        if (job.isPresent() && !job.get().equals(JobState.RUNNING)) {
+        // Exit status check if a job is queued for more than queuedTimeoutMs.
+        if (elapsedTime > queuedTimeoutMs) {
+          if (job.isPresent() && job.get().equals(JobState.QUEUED)) {
+            log.info(
+                "Exit status check: {} due to queued timeout for {}: jobState: {}",
+                getType(),
+                metadata,
+                job.get());
+            break;
+          }
+        }
+        // Exit status check if a job is running for more than taskTimeoutMs.
+        if (elapsedTime > taskTimeoutMs) {
           log.info(
-              "Cancelling job: {} due to timeout for {}: jobState: {}",
+              "Exit status check: {} due to task timeout for {}: jobState: {}",
               getType(),
               metadata,
-              job.get());
-          if (!jobsClient.cancelJob(jobId)) {
-            log.error("Could not cancel job {} for {}", getType(), metadata);
-            return Optional.empty();
-          }
+              job.orElse(null));
           break;
         }
-      }
-      try {
         Thread.sleep(pollIntervalMs);
-      } catch (InterruptedException e) {
-        log.warn(
-            String.format(
-                "Interrupted status polling for job %s for %s. Cancelling the job",
-                getType(), metadata),
-            e);
-        if (!jobsClient.cancelJob(jobId)) {
-          log.error("Could not cancel job {} for {}", getType(), metadata);
-          return Optional.empty();
-        }
       }
+    } catch (InterruptedException e) {
+      // Exit status check if scheduler send out an interrupt signal.
+      log.warn(
+          String.format(
+              "Exit status check: Interrupted status polling for job %s for %s.",
+              getType(), metadata),
+          e);
```

Comment:
Break the loop here with explicit break.

Reply:
It's not in the loop.